ot/gromov/_bregman.py

```diff
@@ -877,6 +1153,307 @@ def entropic_fused_gromov_wasserstein2(
         return logv['fgw_dist']
 
 
+def entropic_BAPG_fused_gromov_wasserstein(
+        M, C1, C2, p=None, q=None, loss_fun='square_loss', epsilon=0.1,
+        symmetric=None, alpha=0.5, G0=None, max_iter=1000, tol=1e-9,
+        marginal_loss=False, verbose=False, log=False):
+    r"""
+    Returns the Fused Gromov-Wasserstein transport between :math:`(\mathbf{C_1}, \mathbf{Y_1}, \mathbf{p})` and :math:`(\mathbf{C_2}, \mathbf{Y_2}, \mathbf{q})`
+    with pairwise distance matrix :math:`\mathbf{M}` between node feature matrices :math:`\mathbf{Y_1}` and :math:`\mathbf{Y_2}`,
+    estimated using Bregman Alternated Projected Gradient method.
+
+    The function solves the following Fused Gromov-Wasserstein
+    optimization problem [63, 64]:
+
+    .. math::
+        \mathbf{T}^* \in\mathop{\arg\min}_\mathbf{T} \quad (1 - \alpha) \langle \mathbf{T}, \mathbf{M} \rangle_F +
+        \alpha \sum_{i,j,k,l} L(\mathbf{C_1}_{i,k}, \mathbf{C_2}_{j,l}) \mathbf{T}_{i,j} \mathbf{T}_{k,l}
+
+        s.t. \ \mathbf{T} \mathbf{1} &= \mathbf{p}
+
+             \mathbf{T}^T \mathbf{1} &= \mathbf{q}
+
+             \mathbf{T} &\geq 0
+    Where :
+
+    - :math:`\mathbf{M}`: metric cost matrix between features across domains
+    - :math:`\mathbf{C_1}`: Metric cost matrix in the source space
+    - :math:`\mathbf{C_2}`: Metric cost matrix in the target space
+    - :math:`\mathbf{p}`: distribution in the source space
+    - :math:`\mathbf{q}`: distribution in the target space
+    - `L`: loss function to account for the misfit between the similarity and feature matrices
+    - :math:`\alpha`: trade-off parameter
+
+    .. note:: By algorithmic design the optimal coupling :math:`\mathbf{T}`
+        returned by this function does not necessarily satisfy the marginal
+        constraints :math:`\mathbf{T}\mathbf{1}=\mathbf{p}` and
+        :math:`\mathbf{T}^T\mathbf{1}=\mathbf{q}`. So the returned Fused
+        Gromov-Wasserstein loss does not necessarily satisfy distance
+        properties and may be negative.
+
+    Parameters
+    ----------
+    M : array-like, shape (ns, nt)
```

Comment:
Tricky situation which is not discussed by authors and simply say that it has to correspond to the pairwise distance feature matrix. However, if `marginal_loss=False` only the crossed-terms should be considered e.g scalar product with an euclidean distance matrix.